File: matlab/examples/basic/cluster/cl_sweep.m
% SF_SWEEP  Arrival-rate sweep on a 2-PS-cluster. Shows response time
% growing as lambda approaches saturation.

clear all

cluster = Cluster().setNumStations(2).setArrivalRate(0.1).setServiceRate(1.0);
cluster.setScheduling(SchedStrategy.PS).setDispatching(RoutingStrategy.RAND);

solverFcn = @(m) MVA(m).getAvgTable();
results = cluster.sweepArrivalRate([0.2, 0.5, 0.9, 1.5], solverFcn);

keys_ = sort(keys(results));
for k = 1:numel(keys_)
    fprintf('\n=== lambda = %g ===\n', keys_(k));
    disp(results{keys_(k)});
end

File: python/examples/basic/cluster/cl_sweep.py
"""
Arrival-rate sweep on a 2-server PS cluster. Demonstrates how response time grows
as the system approaches saturation.
"""

from line_solver import (GlobalConstants, RoutingStrategy, SchedStrategy,
                         Cluster, MVA, VerboseLevel)


if __name__ == "__main__":
    GlobalConstants.set_verbose(VerboseLevel.STD)

    cluster = (Cluster().set_num_stations(2).set_arrival_rate(0.1).set_service_rate(1.0)
            .set_scheduling(SchedStrategy.PS)
            .set_dispatching(RoutingStrategy.RAND))

    rates = [0.2, 0.5, 0.9, 1.5]
    results = cluster.sweep_arrival_rate(rates, MVA)
    for lam, table in results.items():
        print(f"\n=== lambda = {lam} ===")
        print(table)
